pkg/kubelet/cm/cpumanager/policy_static.go

```diff
@@ -310,7 +310,7 @@ func (p *staticPolicy) Allocate(s state.State, pod *v1.Pod, container *v1.Contai
 	}()
 
 	if p.options.FullPhysicalCPUsOnly {
-		CPUsPerCore := p.topology.CPUsPerCore()
+		CPUsPerCore := p.topology.MaxCPUsPerCore()
 		if (numCPUs % CPUsPerCore) != 0 {
```

Comment:
If I'm understanding `MaxCPUsPerCore()` correctly, then it could sometimes return a number that will cause this condition to fail. Is that the desired behaviour? If the CPUs are are taken offline for a legitimate reason, wouldn't we still want to be able to honour the `FullPhysicalCPUsOnly` policy (albeit with fewer CPUs per core).

Reply:
`MaxCPUsPerCore` returns the same result as `CPUsPerCore` , so I think the behaviour would stay the same.
We can consider another similiar scenario. When user specify odd systemReserved CPU num or use reservedCPU option directly, one Core would become 1 CPU available, 1 CPU reserved(just like 1 CPU offline). If requested `numCPUs` above is odd, the condition would still fail.

Reply:
In an ideal (or just better?) world we should have a reliable way to know how many cpus a physical core provides, as reported by the hardware. Unfortunately this is not available from cadvisor. In this case (except the numa-distributed case we discuss below) it seems to me we have a good approximation. It may still fail in some pathological corner cases I think, for example if all but one SMT pairs have offlined CPUs, but it seems still better than before.